ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -382,12 +381,33 @@ S3CustomizeFileSystem(WT_STORAGE_SOURCE *storageSource, WT_SESSION *session, con
     std::string objPrefix;
     if ((ret = s3->wtApi->config_get_string(
            s3->wtApi, session, config, "prefix", &objPrefixConf)) == 0)
-        objPrefix = objPrefixConf.str;
+        objPrefix = std::string(objPrefixConf.str, objPrefixConf.len);
     else if (ret != WT_NOTFOUND) {
         std::cerr << "Error: customize_file_system: config parsing for object prefix";
         return (1);
     }
 
+    /* Configure the AWS Client configuration. */
+    Aws::S3Crt::ClientConfiguration awsConfig;
+    awsConfig.throughputTargetGbps = throughputTargetGbps;
+    awsConfig.partSize = partSize;
+
+    /*
+     * Get the AWS region to be used. The allowable values for AWS region are listed here in the AWS
+     * documentation: http://sdk.amazonaws.com/cpp/api/LATEST/namespace_aws_1_1_region.html
+     */
+    WT_CONFIG_ITEM regionConf;
+    std::string region;
+    if ((ret = s3->wtApi->config_get_string(s3->wtApi, session, config, "region", &regionConf)) ==
+      0)
+        awsConfig.region = std::string(regionConf.str, regionConf.len);
+    else if (ret != WT_NOTFOUND) {
+        std::cerr << "Error: customize_file_system: config parsing for AWS region";
+        return (1);
+    } else {
```

Comment:
@sulabhM I added another clause to the if-statement here to differentiate between the case when no region has been provided, and when `config_get_string` actually fails for some other reason. Is this appropriate? Or would it be better to collapse these two paths into a single statement?